src/coord/src/coord.rs

```diff
@@ -2066,10 +2136,23 @@ where
                 catalog::OpStatus::DroppedItem(entry) => {
                     self.report_catalog_update(entry.id(), entry.name().to_string(), -1);
                     match entry.item() {
-                        CatalogItem::Table(_) | CatalogItem::Source(_) => {
+                        item @ CatalogItem::Table(_) | item @ CatalogItem::Source(_) => {
                             sources_to_drop.push(entry.id());
+                            self.update_mz_columns_catalog_view(
+                                item.desc().unwrap(),
+                                &entry.name().to_string(),
+                                entry.id(),
+                                -1,
+                            );
+                        }
+                        CatalogItem::View(catalog::View { desc, .. }) => {
+                            self.update_mz_columns_catalog_view(
+                                desc,
+                                &entry.name().to_string(),
+                                entry.id(),
+                                -1,
+                            );
```

Comment:
Similar in spirit to the last comment; I'd stick a `if let Ok(desc) = item.desc()` _after_ the match to handle all the things that need to remove from the `mz_columns` view in one fell swoop.